packages/components/src/validated-form-controls/control-with-error.tsx

```diff
@@ -101,17 +101,65 @@ function UnforwardedControlWithError< C extends React.ReactElement >(
 	>();
 	const [ isTouched, setIsTouched ] = useState( false );
 	const previousCustomValidityType = usePrevious( customValidity?.type );
+	const className = 'components-validated-control';
 
-	// Ensure that error messages are visible after user attemps to submit a form
-	// with multiple invalid fields.
 	useEffect( () => {
 		const validityTarget = getValidityTarget();
+
+		// Ensure that error messages are visible after user attemps to submit a form
+		// with multiple invalid fields.
 		const showValidationMessage = () =>
 			setErrorMessage( validityTarget?.validationMessage );
 
+		// Suppress the native error popover, while keeping the focus behavior intact.
+		const supressNativePopover = ( event: Event ) => {
+			event.preventDefault();
+
+			const target = event.target as ValidityTarget;
+
+			if ( ! target.form ) {
+				target.focus();
+				return;
+			}
+
+			const firstErrorInForm = Array.from( target.form.elements ).find(
+				( el ) => ! ( el as ValidityTarget ).validity.valid
+			);
+
+			if ( firstErrorInForm === target ) {
+				target.focus();
+			}
+		};
+
+		// Radio inputs need special handling because all radio inputs with the
+		// same `name` will be marked as invalid.
+		const radioSibilings =
```

Comment:
One of the topics discussed in the original issue was whether this would be "worth the extra trouble".

So far, from the set of use cases that we currently have, radio inputs were the only thing that needed special handling like this. There may be more, and I have some qualms about the screen reader behavior, but perhaps not too bad to count as hard blockers.

Reply:
My bad. Here's what happens when you click the submit button on an untouched radio group. The "Before" version has the radio handling in this hook commented out.

| Before | After |
|--------|-------|
|<img width="1084" height="802" alt="Radio group, before" src="https://github.com/user-attachments/assets/96fe0415-9744-4db0-9510-9b052f7cb322" />|<img width="1084" height="802" alt="Radio group, after" src="https://github.com/user-attachments/assets/720d9e3d-f59d-4402-b74e-c5dfe9f08101" />|


I expanded on the code comment a bit (793ace03a6d7625649297f5880fa3045870f389d). Ideally we have a unit test, but unfortunately this is not accurately testable in jsdom.